splinterd/src/main.rs

```diff
@@ -460,33 +468,63 @@ fn main() {
+                root: RootConfig {
+                    appenders: vec![String::from("default")],
+                    filter: log::Level::Info,
+                },
+                appenders: vec![AppenderConfig {
+                    name: String::from("default"),
+                    encoder: default_pattern(),
+                    target: LogTarget::StdOut,
+                }],
+                loggers: vec![],
+            };
+            let log_handle = log4rs::init_config(
+                //Always produces a valid config try_into_config just returns a Result
+                default_config
+                    .try_into()
+                    .expect("Default logging config failure"),
+            );
+            if let Err(e) = log_handle {
+                eprintln!("Could not start logging, {}", e);
+                std::process::exit(1);
+            }
+            log_handle.unwrap()
+        }
+        #[cfg(not(feature = "log-config"))]
+        {
+            let encoder =
+                PatternEncoder::new("[{d(%Y-%m-%d %H:%M:%S%.3f)}] T[{T}] {l} [{M}] {m}\n");
+            let stdout = log4rs::append::console::ConsoleAppender::builder()
+                .encoder(Box::new(encoder))
+                .build();
+            let config = log4rs::Config::builder()
+                .appender(Appender::builder().build("stdout", Box::new(stdout)))
+                .logger(Logger::builder().build("hyper", log::LevelFilter::Warn))
+                .logger(Logger::builder().build("tokio", log::LevelFilter::Warn));
+            #[cfg(feature = "https-bind")]
+            let config = config.logger(Logger::builder().build("h2", log::LevelFilter::Warn));
+            let conf = config.build(
+                Root::builder()
+                    .appender("stdout")
+                    .build(get_log_filter_level(&matches)),
+            );
 
-    let encoder = PatternEncoder::new("[{d(%Y-%m-%d %H:%M:%S%.3f)}] T[{T}] {l} [{M}] {m}\n");
-    let stdout = log4rs::append::console::ConsoleAppender::builder()
-        .encoder(Box::new(encoder))
-        .build();
-    let config = log4rs::Config::builder()
-        .appender(Appender::builder().build("stdout", Box::new(stdout)))
-        .logger(Logger::builder().build("hyper", log::LevelFilter::Warn))
-        .logger(Logger::builder().build("tokio", log::LevelFilter::Warn));
-    #[cfg(feature = "https-bind")]
-    let config = config.logger(Logger::builder().build("h2", log::LevelFilter::Warn));
-    let log_config = config.build(Root::builder().appender("stdout").build(log_level));
-    if let Ok(lc) = log_config {
-        let _handle = log4rs::init_config(lc);
-    } else {
-        unreachable!();
-        // log_config  should always return an Ok variant.
-        // Err is returned if multiple logger facades are registered which shouldn't ever happen
-    }
+            if let Ok(lc) = conf {
+                log4rs::init_config(lc).unwrap()
```

Comment:
Remove the unwrap calls, here and elsewhere

Reply:
No more just raw unwrap calls